### PR TITLE
Fix i386 bugs

### PR DESCRIPTION
**Pull request by ParkHanbum**
this PR have fix some bugs in i386.
include #762 #805.

### CODE REVIEW

File: check-deps/__have_libcapstone.c [hunk 1]
```diff
@@ -8,6 +8,6 @@
 int main()
 {
 	cs_insn insn;
-	printf("size: %lu\n", sizeof(insn));
+	printf("size: %d\n", sizeof(insn));
```
Comment (namhyung) [inline check-deps/__have_libcapstone.c]: `%zu` might be a better fit.  See [man 3 printf](https://linux.die.net/man/3/printf) for "The length modifier".

File: utils/symbol.c [hunk 2]
```diff
@@ -201,7 +201,7 @@ static void unload_symtab(struct symtab *symtab)
 }
 
 static int load_symbol(struct symtab *symtab, unsigned long prev_sym_value,
-		       unsigned long offset, unsigned long flags,
+		       unsigned long long offset, unsigned long flags,
```
Comment (namhyung) [inline utils/symbol.c]: I don't get it.  Could you please provide a concrete example?

File: utils/symbol.c [hunk 3]
```diff
@@ -1600,6 +1608,7 @@ TEST_CASE(symbol_load_map) {
 	struct symtabs symtabs = {
 		.dirname = "",
 		.kernel_base = -4096ULL,
+		.flags = SYMTAB_FL_ADJ_OFFSET,
```
Comment (namhyung) [inline utils/symbol.c]: Basically non-PIE has same issue and I think it works as is.  Can you verify this change doesn't break other arch?

File: tests/t223_dynamic_full.py [hunk 4]
```diff
@@ -19,9 +19,13 @@ def pre(self):
         return TestBase.TEST_SUCCESS
 
     def build(self, name, cflags='', ldflags=''):
+        import platform
+        cflags += ' -fno-pie'
+        if not platform.machine().startswith('i686'):
+            cflags += ' -fno-plt'  # workaround of build failure
```
Comment (namhyung) [inline tests/t223_dynamic_full.py]: I think BI (build) failure is ok to have.  It just says current environment doesn't provide such feature and we can let it fail.

Comment (ParkHanbum) [reply tests/t223_dynamic_full.py]: @namhyung ok, then, I'll update this when full-dynamic able.

File: utils/symbol.c (hunk 3 above)
Comment (ParkHanbum) [reply utils/symbol.c]: @namhyung x86_64 was ok with this. but I did not test it on the Arm base.

File: utils/symbol.c (hunk 2 above)
Comment (ParkHanbum) [reply utils/symbol.c]: I'll update detail of this.

File: check-deps/__have_libcapstone.c (hunk 1 above)
Comment (ParkHanbum) [reply check-deps/__have_libcapstone.c]: updated!

File: utils/symbol.c (hunk 2 above)
Comment (ParkHanbum) [reply utils/symbol.c]: @namhyung I described it at commit message. please refer here : b9e7fcdd6f4f1043f07d9b57979558a40910eb71

**Comment by ParkHanbum** [comment]
@namhyung could you review this again plz?

File: utils/symbol.c (hunk 3 above)
Comment (namhyung) [reply utils/symbol.c]: OK, then let's change it like below:

```c
for (i = 0; i < info->dlpi_phnum; i++) {
    if (info->dlpi_phdr[i].p_type != PT_LOAD)
        continue;
    map->start = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
    map->end = map->start + info->dlpi_phdr[i].p_memsz;
    break;
}
```

File: utils/symbol.c (hunk 2 above)
Comment (namhyung) [reply utils/symbol.c]: Thanks for the explanation

File: utils/symbol.c (hunk 3 above)
Comment (ParkHanbum) [reply utils/symbol.c]: It was applied! test passed at i386 and x64

**Comment by ParkHanbum** [comment]
@namhyung this PR has been updated!

**Comment by honggyukim** [comment]
@ParkHanbum Please rebase it to master before updating the PR.

**Comment by honggyukim** [comment]
It has a compilation error as follows:
```
$ ARCH=i386 CFLAGS=-m32 LDFLAGS=-m32 ./configure

$ make
  CC       arch/i386/cpuinfo.o
  LINK     arch/i386/uftrace.o
  LINK     uftrace
  ASM      arch/i386/mcount.op
  ASM      arch/i386/common.op
  ASM      arch/i386/fentry.op
  ASM      arch/i386/plthook.op
  CC FPIC  arch/i386/mcount-dynamic.op
/home/honggyu/work/uftrace/arch/i386/mcount-dynamic.c:105:5: error: conflicting types for ‘mcount_patch_func’
 int mcount_patch_func(struct mcount_dynamic_info *mdi, struct sym *sym,
     ^
In file included from /home/honggyu/work/uftrace/arch/i386/mcount-dynamic.c:9:0:
/home/honggyu/work/uftrace/libmcount/internal.h:434:5: note: previous declaration of ‘mcount_patch_func’ was here
 int mcount_patch_func(struct mcount_dynamic_info *mdi, struct sym *sym,
     ^
```
The following change is needed to compile this PR. It's not a fault of @ParkHanbum but just missed to update `i386` code in 2fa2ff8d.
```diff
diff --git a/arch/i386/mcount-dynamic.c b/arch/i386/mcount-dynamic.c
index 4875909..ce5896b 100644
--- a/arch/i386/mcount-dynamic.c
+++ b/arch/i386/mcount-dynamic.c
@@ -103,7 +103,7 @@ static int patch_fentry_func(struct mcount_dynamic_info *mdi, struct sym *sym)
 }
 
 int mcount_patch_func(struct mcount_dynamic_info *mdi, struct sym *sym,
-                     struct mcount_disasm_engine *disasm)
+                     struct mcount_disasm_engine *disasm, unsigned min_size)
 {
        return patch_fentry_func(mdi, sym);
 }
```
I just tested it and #762 is fixed by this.

**Comment by ParkHanbum** [comment]
@honggyukim thanks you for letting me know. I updated it!

**Comment by honggyukim** [comment]
It works fine now. Thanks!

**Comment by honggyukim** [comment]
But please remove the useless line at the end of the commit message.
```
commit 0f63e2e917116cae24989d4969e9c466f7d9ab91
Author:     Hanbum Park <kese111@gmail.com>
AuthorDate: Wed Jun 12 00:52:51 2019 -0700

    build: Fix a build error for i386
    
    There is a format string in libcapstone.c that did not match the type.
    because of it, configuration was not done as intended. to fix this
    issue, current format string "%lu" was changed to "%zu".
    
    function 'mcount_patch_func' declaration was changed after size filter
    updated. so, to solve conflict its declaration inside of i386 arch also
    changed.
    
    Fixed : #805
    
    Signed-off-by: Hanbum Park <kese111@gmail.com>
    
    merge to build
```
`merge to build` has to be removed.

**Comment by ParkHanbum** [comment]
@honggyukim my mistake..! thank you!

**Pull request merged**